src/prefect/_internal/concurrency/services.py

```diff
@@ -0,0 +1,139 @@
+import abc
+import asyncio
+import atexit
+import threading
+from collections import deque
+from typing import Generic, Optional, TypeVar
+
+from prefect._internal.concurrency.api import create_call, from_sync
+from prefect._internal.concurrency.calls import Call
+from prefect._internal.concurrency.event_loop import call_soon_in_loop
+from prefect._internal.concurrency.threads import get_global_loop
+
+T = TypeVar("T")
+
+
+class QueueService(abc.ABC, Generic[T]):
+    _instance = None
+    _instance_lock = threading.Lock()
+
+    def __init__(self) -> None:
+        self._queue: Optional[asyncio.Queue] = None
+        self._loop: Optional[asyncio.AbstractEventLoop] = None
+        self._done_event: Optional[asyncio.Event] = None
+        self._task: Optional[asyncio.Task] = None
+        self._early_items = deque()
+        self._lock = threading.Lock()
+        self._stopped: bool = False
+        self._started: bool = False
+
+    def start(self):
+        self._queue = asyncio.Queue()
+        self._loop = asyncio.get_running_loop()
+        self._done_event = asyncio.Event()
+        self._task = self._loop.create_task(self._run())
+        self._started = True
+        self._start_call: Optional[Call] = None
+
+        # Put all early submissions in the queue
+        while self._early_items:
+            self.send(self._early_items.pop())
+
+        # Stop at interpreter exit by default
+        atexit.register(self.drain)
+
+    def stop(self):
+        # Stop sending work to this instance
+        self._clear_instance()
```

Comment:
This should send `self` so that it only clears the instance _if_ this was the current one :)